subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/TestBuildCache.groovy

```diff
@@ -70,6 +70,17 @@ class TestBuildCache {
         cacheDir.listFiles().findAll { it.name ==~ /\p{XDigit}{${Hashing.defaultFunction().hexDigits}}/ }.sort()
     }
 
+    TestFile getCacheFile(String cacheKey) {
+        cacheDir.listFiles().find { it.name == cacheKey }
+    }
+
+    /**
+     * TODO: Use local build cache operations to check if the cache file was stored, once they are implemented
+     */
+    boolean hasCacheFile(String cacheKey) {
+        cacheDir.listFiles().any { it.name == cacheKey }
+    }
+
```

Comment:
🤔 This is already an improvement, so yay!

However, I think it would be worth to take one more step here, and divorce `TestBuildCache` from the idea of file based storage. We could have `hasCacheEntry(String cacheKey)`, And we could have something like `changeCacheEntryContent(Function<String, String> transformer)` and `String getCacheEntryContent()`.

Reply:
I introduced `TestCacheEntry` interface, that could be implemented with different backed cache entries.

All single entry queries now return `TestCacheEntry`. There are still other methods that are still file aware, like:
```
List<TestFile> listCacheFiles() 
List<TestFile> listCacheTempFiles()
List<TestFile> listCacheFailedFiles()
```

These require a bit more work to replace with TestCacheEntry. So maybe yeah we should do that as a part of some other PR.